frontend/src/metabase/query_builder/actions/querying.ts

```diff
@@ -222,18 +238,20 @@ export const queryErrored = createThunkAction(
       if (error && error.isCancelled) {
         return null;
       } else {
-        return { error: error, duration: new Date() - startTime };
+        return { error: error, duration: Date.now() - startTime };
       }
     };
   },
 );
 
 export const CANCEL_QUERY = "metabase/qb/CANCEL_QUERY";
-export const cancelQuery = () => (dispatch, getState) => {
+export const cancelQuery = () => (dispatch: Dispatch, getState: GetState) => {
   const isRunning = getIsRunning(getState());
   if (isRunning) {
     const { cancelQueryDeferred } = getState().qb;
     if (cancelQueryDeferred) {
+      // it could have type Deferred + resolve/reject, please fix other similar places
+      // @ts-expect-error probably it never worked, promise.resolve() doesn't exist in spec
```

Comment:
Shouldn't we update the type of `cancelQueryDeferred` to `Deferred | Promise<void> | null`?

Reply:
probably, but I'm not sure